src/twisted/python/compat.py

```diff
@@ -150,6 +151,22 @@
 )
 
 
+def md5(data=b"", **kwargs):
+    """
+    Wrapper around hashlib.md5
+    Attempt call with 'usedforsecurity=False' if we get a ValueError, which happens when
+    OpenSSL FIPS mode is enabled:
+    ValueError: error:060800A3:digital envelope routines:EVP_DigestInit_ex:disabled for fips
+    """
+
+    try:
+        return hashlib.md5(data, **kwargs)
+    # the below value error will be hit only when fips is enabled
+    # so skip it while testing for code coverage
+    except ValueError:  # pragma: no cover
+        return hashlib.md5(data, **kwargs, usedforsecurity=False)  # pragma: no cover
```

Comment:
I think that from the FIPS mode, we should run the tests with coverage and report the ccoverage...

...but my question :)

If we always ignore FIPS mode, why don't we always call md5 with `useforsecurity=False`

I think is good to have all md5 usage in a dedicated function.
Later we can raise an exception with an useful error message... like "Failure. FIPS mode is enabled, and you are trying to use MD5 code"

Reply:
> If we always ignore FIPS mode, why don't we always call md5 with useforsecurity=False

Good question. `usedforsecurity` is available only in `python >= 3.9` hence this is needed for now.